diff --git a/meson.build b/meson.build
index df1f291b..96170fe9 100644
--- a/meson.build
+++ b/meson.build
@@ -286,6 +286,12 @@ elif host_machine.system() == 'menix'
 	internal_conf.set10('MLIBC_MMAP_ALLOCATE_DSO', true)
 	internal_conf.set10('MLIBC_MAP_FILE_WINDOWS', true)
 	subdir('sysdeps/menix')
+elif host_machine.system() == 'template'
+	disable_linux_option = true
+	static_pic_enabled = false
+	rtld_include_dirs += include_directories('sysdeps/template/include')
+	libc_include_dirs += include_directories('sysdeps/template/include')
+	subdir('sysdeps/template')
 # ANCHOR: demo-sysdeps
 elif host_machine.system() == 'demo'
 	subdir('sysdeps/demo')
diff --git a/options/rtld/generic/main.cpp b/options/rtld/generic/main.cpp
index d4660cb5..d5dd9ec0 100644
--- a/options/rtld/generic/main.cpp
+++ b/options/rtld/generic/main.cpp
@@ -541,11 +541,6 @@ extern "C" void *interpreterMain(uintptr_t *entry_stack) {
 #else
 	(void)env_ld_library_path;
 	(void)env_ld_preload;
-	auto ehdr = reinterpret_cast<elf_ehdr*>(__ehdr_start);
-	phdr_pointer = reinterpret_cast<void*>((uintptr_t)ehdr->e_phoff + (uintptr_t)ehdr);
-	phdr_entry_size = ehdr->e_phentsize;
-	phdr_count = ehdr->e_phnum;
-	entry_pointer = reinterpret_cast<void*>(ehdr->e_entry);
 #endif
 	__ensure(phdr_pointer);
 	__ensure(entry_pointer);
